### ISSUE TITLE
Optional "group_count" attribute for user-defined type converters

### ISSUE THREAD

**Issue by jenisys** [r1chardj0n3s/parse#18]
The current "parse" module has as small deficiency (or bug).
When user-defined type converter uses regular expression grouping in its pattern (attribute), the extracted result parameters are partly wrong in some params because this group index offset is not considered.
NOTE: This problem occurs only for fixed (unnamed) fields, named fields are OK.

``` python
# FILE: parse.py
# NECESSARY CHANGES:
…
        # -- Parser._handle_field()
        ...
        if type in self._extra_types:
            type_converter = self._extra_types[type]
            s = getattr(type_converter, 'pattern', r'.+?')
            # -- EXTENSION: group_count attribute
            group_count = getattr(type_converter, 'group_count', 0)
            self._group_index += group_count
            # -- EXTENSION-END
```

**Comment by r1chardj0n3s** [comment]
I know this is a very old bug, is it still an issue?

**Comment by jenisys** [comment]
That was fixed by adding `regex_group_count` in parse-1.8.3.

SEE ALSO:
* [parse.py:968](https://github.com/r1chardj0n3s/parse/blob/487d8762b41c0306df2e7fd1f9adf2b4b25986ff/parse.py#L968)
